### ISSUE TITLE
Polymer 2.x/ Webcomponents 1.0 support

### ISSUE THREAD

**Issue by ergo** [expandjs/mat-elements#4]
Hello,
The demos and elements are running under polymer 1.x, will there be a transition to Polymer 2.0 at some point of time?

**Comment by Pupix** [comment]
Hello,
Currently there's no plan for a 2.0 upgrade. Upgrading for Polymer 2.0 is not as easy as the Polymer team claimed it will be ([even their own devs are complaining](https://github.com/Polymerelements/paper-input/blob/master/paper-input.html#L165)), also Polymer 3.0 is on his way with an experimental, React like, HTML handling with template-literals ([lit-html](https://github.com/PolymerLabs/lit-html)).

We feel that updating at this time wouldn't be beneficial and we might have to start rewriting everything again once Polymer 3.0 hits.

Cheers.

**Comment by ergo** [comment]
I actually updated my app to 2.0 using hybrid and non-hybrid mode. It wasn't nearly as bad as it might look ;-) I already saw that you guys use slots so apart the observers working differently (any value needs to be defined for observer to fire), most elements should work already with legacy mixin or hybrid mode.

Btw. 3.0 is supposed to be 100% API compatible, so no work on your end should be needed to support that - the translation to template strings will be handled by build step, same as you would do with current webpack polymer plugin.
